src/ring_theory/ideal_operations.lean

```diff
@@ -538,6 +538,26 @@ end map_and_comap
 
 end ideal
 
+namespace is_ring_hom
+
+variables {R : Type u} {S : Type v} [comm_ring R] [comm_ring S]
+variables (f : R → S) [is_ring_hom f]
+
+def ker : ideal R := ideal.comap f ⊥
+
+@[simp] lemma ker_eq : ((ker f) : set R) = is_add_group_hom.ker f := rfl
+
+lemma inj_of_trivial_ker (h : ker f = ⊥) : function.injective f :=
+is_add_group_hom.inj_of_trivial_ker f $ by rw [←ker_eq, h]; refl
+
+lemma trivial_ker_of_inj (h : function.injective f) : ker f = ⊥ :=
+submodule.ext' $ by rw [ker_eq, is_add_group_hom.trivial_ker_of_inj f h]; refl
+
+lemma inj_iff_trivial_ker : function.injective f ↔ ker f = ⊥ :=
```

Comment:
It suffices to have this iff-lemma. You can remove the two implication-lemmas above it.
On the other hand, I would like to have the lemma that you only need to prove that `f x = 0 → x = 0`. That is usually an easier condition than actual injectivity.

Reply:
Should I just add a lemma, `ker_eq_bot_iff_eq_zero` maybe, saying `ker f = ⊥ ↔ ∀ x, f x = 0 → x = 0`?

Reply:
I suppose that it should be `ker_eq_bot` instead of `trivial_ker` too.

Reply:
Yes please make those two lemmas.